### ISSUE TITLE
LSTM (Univariate) using only production_cost_total

### ISSUE THREAD

**Issue by FNUAreena** [FNUAreena/fall-2025-group9#74]
**Goal:**
Train an LSTM model using only `production_cost_total` (one feature) to learn how production cost changes over time for each school and meal type, and check how well it fits past data.

**Data :** 
- Uses the file: `meals_combined.csv`
- Looks at each school's breakfast and lunch separately
- Predicts: Total production cost per day

**LSTM Model**
- The LSTM is defined in the `ForecastingModel` class in  
  [model.py](https://github.com/FNUAreena/fall-2025-group9/blob/main/src/component/univariate/model.py).
- Main settings:
  - `MODEL_TYPE = "LSTM"`
  - `INPUT_DIM = 1`, `OUTPUT_DIM = 1`
  - `HIDDEN_DIM = 256`
  - `NUM_LAYERS = 4`
  - `DROPOUT = 0.25`
- The model runs on GPU if available, otherwise on CPU.

**Training:** 

- The training loop is in  
  [main.py](https://github.com/FNUAreena/fall-2025-group9/blob/main/src/component/univariate/main.py).
- Hyperparameters:
  - `EPOCHS = 100`
  - `BATCH_SIZE = 64`
  - Learning rate `LR = 0.001`
  - Fixed random seed `SEED = 42` for reproducible results.
- For each `(school_name, meal_type)` with enough data:
  - Build train and test `DataLoader`s.
  - Create an LSTM model with `ForecastingModel("LSTM", ...)`.
  - Train for 100 epochs using:
    - Optimizer: Adam
    - Loss: Mean Squared Error (MSE)
- After training each model, we save it under  
  `univariate/LSTM_models/` with a name like  
  `LSTM_<school_name>_<meal_type>.pth`.

Here we have saved our models in google drive due to high storage [Lstm models](https://drive.google.com/drive/folders/1daVVcrvS2u6rFBcKG_RjkQJh6Q19z4P7?usp=drive_link)

**Evaluation:** 
- For every trained model, we:
  - Predict on the test windows.
  - Convert predictions and true values back to original currency units
    (inverse transform of the scaler).
  - Add them to two global lists: `all_true` and `all_pred`.
- After all schools and meals are processed, we compute overall metrics:
  - **MSE**: `18793.44`
  - **RMSE**: `137.08`
  - **R²**: `0.86`
- We also create a **scatter plot**:
  - X-axis: actual production cost.
  - Y-axis: LSTM predicted production cost.
  - A dashed `y = x` line shows perfect prediction.
  - Title: `LSTM Predicted vs Actual`.
  - Saved as `univariate/plots/LSTM.png` in the repo  
    (file: [LSTM.png](https://github.com/FNUAreena/fall-2025-group9/blob/main/src/component/univariate/plots/LSTM.png)).

**How to run :** 
```bash
cd src/component 
python  univariate/main.py
